release-notes.txt

```diff
@@ -1,3 +1,688 @@
+    * sys/xtimer: concurrency/robustness improvements
+    * sys/xtimer/xtimer.c: _mutex_timeout() fix
+    - sys/xtimer: Remove dependency to core_msg
+
+Networking
+----------
+
+    + sys/gnrc_lorawan: add initial support for GNRC based LoRaWAN stack
+    * sys/gnrc_netif: make 6LoENC dynamically configurable
+    + sys/gnrc_sock: provide asynchronous event implementation
+    * sys/gnrc_ipv6: fix source check for loopback address
+    * sys/gnrc_ipv6_ext_frag: remove fragment header when n-th fragment is first
+    * sys/gnrc_ipv6_nib: allow for configuration of static link-local addresses
+    * sys/gnrc_netif: highest source address scope wins selection
+    * sys/gnrc_netif: only use prefix matching as tie-breaker in source selection
+    * sys/gnrc_sixlowpan_frag_rb: separate out classic frag specific code
+    * sys/gnrc_sixlowpan_frag_rb: fix memory-leak in _rm_by_datagram() and in interval marker inherited from base
+    * sys/gnrc_sixlowpan_frag: various optimizations on sending
+    * sys/gnrc_sixlowpan_iphc: add fragment forwarding stubs 
+    * sys/gnrc_sixlowpan_iphc_nhc: determine UDP hdr length from RB, fix fragmented data size allocation
+    * sys/gnrc_tcp: return immediately on gnrc_tcp_recv if a connection is closing 
+    + sys/net/gcoap: add macro to delay initialization of gcoap
+    * sys/net/gcoap: do not allocate RX buf on stack
+    * sys/pthread: check return value of malloc(), prevent NULL pointer deference
+    * sys/shell/gnrc_netif: Use netif API for ifconfig
+    * sys/shell/sc_nimble`_netif: allow connecting by name
+    + sys/sock: initial definitions for asynchronous event handling
+    + sys/sock_async: initial import of event-based implementation
+
+Packages
+--------
+    * pkg: cleanup management of dependencies
+    * pkg/pkg.mk: use intermediate state files 
+    * pkg/ccn-lite: bump version
+    + pkg/flatbuffers: add support for FlatBuffers serialization library
+    * pkg/gecko_sdk: update to version 2.7
+    * pkg/libfixmath: several improvements and fixes to support 8bit
+    * pkg/littlefs: bump littlefs version to 1.7.2
+    * pkg/lora-serialization: bump to latest version and update test for AVR
+    * pkg/lwip: add support for esp32 Ethernet device
+    * pkg/lwip: add IPv4 support for LWIP-stm32
+    * pkg/lwip: start DHCP for a netif with lwip_dhcp
+    * pkg/lwip: add STM32 support
+    * pkg/lwip: enable lwIP for ESP8266
+    + pkg/nanopb: add nanoPb protocol buffers library package
+    * pkg/nimble: bump version
+    + pkg/nimble: add simple BLE connection manager: autoconn
+    + pkg/nimble/autoconn: add user event callback
+    + pkg/nimble_netif: catch L2CAP connection failures
+    + pkg/nimble_netif: add additional events
+    * pkg/tinydtls: add DTLS sock API implementation
+    * pkg/tinydtls: bump version
+    + pkg/tensorflow-lite: add support to RIOT
+    + pkg/utensor: add support to RIOT
+    * pkg/wakaama: Add basic LWM2M client implementation
```

Comment:
Add: `* pkg/u8g2: refactor the U8g2 package`

(https://github.com/RIOT-OS/RIOT/commit/b253f61a7f3522b018fdbbd8d55f597fc06e61e7)